source/shared/cpp/ObjectModel/TextBlock.cpp

```diff
@@ -145,6 +149,41 @@ void TextBlock::SetHorizontalAlignment(const HorizontalAlignment value)
     m_hAlignment = value;
 }
 
+void TextBlock::SetLanguage(const std::locale& value)
+{
+    m_language = value;
+}
+
+std::wstring TextBlock::StringToWstring(const std::string& in) const
+{
+#ifdef _WIN32
+    std::wstring_convert<std::codecvt_utf8_utf16<wchar_t>, wchar_t> utfConverter;
+    return utfConverter.from_bytes(in);
+#else
+    std::wstring_convert<std::codecvt_utf8_utf16<wchar_t>, wchar_t> utfConverter;
+    return utfConverter.from_bytes(input);
```

Comment:
>input [](start = 35, length = 5)

in?  Can you give me the background on why we need these ifdef'd versions?

Reply:
The wstrings differ by size in different platforms, in ios, Linux and android the wstring characters are 4 bytes in length, while the windows ones are 2 bytes long. 
https://stackoverflow.com/questions/7232710/convert-between-string-u16string-u32string

Reply:
In this function these look the same except for the parameter name...

---
In reply to: [166771554](https://github.com/Microsoft/AdaptiveCards/pull/1102#discussion_r166771554) [](ancestors = 166771554)